### PR TITLE
reel: migrate old tokens and update link flow to match

### PR DESCRIPTION
**Pull request by arthyn**
OTT, we only migrated tokens in %bait, but we need to migrate them here in %reel so that both the old UI and the new UI can work.

### CODE REVIEW

File: desk/app/reel.hoon [hunk 1]
```diff
@@ -174,18 +188,20 @@
...
     ~[[%pass [%token-link-want token ~] %agent [src dap]:bowl %poke %reel-give-token-link !>(result)]]
       %reel-give-token-link
     =+  !<(result=(unit [cord cord]) vase)
     ?~  result  `this
     :_  this
     =/  [token=cord url=cord]  u.result
-    ?:  =('' url)
-      ~[[%give %fact ~[[%token-link (scot %p src.bowl) token ~]] %json !>(~)]]
-    ~[[%give %fact ~[[%token-link (scot %p src.bowl) token ~]] %json !>(s+url)]]
+    =/  path  (stab (cat 3 '/token-link/' token))
```
Comment (Fang-) [inline desk/app/reel.hoon:203]: Some smell on this not being `/token-link/(scot %t token)`, but iirc that's for backcompat reasons that are out of scope here, right?

Comment (Fang-) [reply desk/app/reel.hoon:203]: (Also some of the `cord`s in the surrounding code should probably be `token:reel`s.)

File: desk/app/reel.hoon [hunk 2]
```diff
@@ -79,7 +80,20 @@
...
+        ~(tap by our-metadata)
+      |=  [[=token:reel =metadata:reel] [md=_our-metadata id=_stable-id]]
+      ?^  (slaw %uv token)  [md id]
+      ?^  (rush token flag)
+        :-  md
+        ?:  (~(has by id) token)  id
+        (~(put by id) token token)
+      =/  new  (rap 3 (scot %p our.bowl) '/' token ~)
+      :-  (~(put by md) new metadata)
+      (~(put by id) new new)
```
Comment (Fang-) [inline desk/app/reel.hoon:95]: This is fine to run on every load, because none of the transformations result in follow-up:
- New-style tokens (`@uv` format) are untouched,
- "New old-style" tokens (`~ship/term` format) only guarantee they have an entry in `stable-id`,
- "Old old-style" tokens get upgraded into "new old-style" tokens and put an entry in `stable-id`.

Correct?

**Review by Fang- — APPROVED** [review]
I think this is good, on the condition that my comments check out.

File: desk/app/reel.hoon (hunk 2 above)
Comment (Fang-) [reply desk/app/reel.hoon:95]: (I'd still feel more comfortable with migration moving into state version `%5`, fwiw, just because it doesn't even risk repeated transforms.)

File: desk/app/reel.hoon (hunk 1 above)
Comment (arthyn) [reply desk/app/reel.hoon:203]: hmm this should be able to work I think

File: desk/app/reel.hoon (hunk 2 above)
Comment (arthyn) [reply desk/app/reel.hoon:95]: yep!

File: desk/app/reel.hoon (hunk 1 above)
Comment (arthyn) [reply desk/app/reel.hoon:203]: nvm we can't do this
